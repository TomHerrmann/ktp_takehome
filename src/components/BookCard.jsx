import React from 'react';

const BookCards = ({ book }) => {
  const { title, author, publisher, publishDate } = book;

  return (
    <article className="book-card">
      <h2>Title: {title}</h2>
      <ul>
        <li>
          <strong>Author: </strong>
          {author}
        </li>
        <li>
          <strong>Publisher: </strong>
          {publisher}
        </li>
        <li>
          <strong>Publish Date: </strong>
          {publisher}
        </li>
      </ul>
    </article>
  );
};

export default BookCards;
