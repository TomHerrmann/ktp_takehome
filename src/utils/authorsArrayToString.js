const authorsArrayToString = (authorsArray) => {
  return authorsArray.reduce((acc, curr) => {
    acc += `${curr}, `;
    return acc;
  }, '');
};

export default authorsArrayToString;
