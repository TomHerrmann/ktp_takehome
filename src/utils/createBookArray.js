const createBookArray = (arr) => {
  return arr.map((item) => {
    const { title, authors, publisher, publishedDate } = item.volumeInfo;

    // creates alternative value to cover books without listed publishers
    return { title, authors, publisher: publisher || 'N/A', publishedDate };
  });
};

export default createBookArray;
