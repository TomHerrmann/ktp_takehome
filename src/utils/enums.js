// api enums

export const apiURL =
  'https://www.googleapis.com/books/v1/volumes?q=kaplan%20test%20prep';

// test enums

export const unformattedBooksArray = [
  {
    volumeInfo: {
      title: "Tom's Test Literature",
      authors: ['Chris', 'Mary'],
      publisher: "Tom's Books",
      publishedDate: '2020-06-11',
      other: 'extra stuff',
    },
  },
  {
    volumeInfo: {
      title: "Tom's SAT Prep",
      authors: ['Chris', 'Mary'],
      publisher: "Tom's Books",
      publishedDate: '2020-06-11',
      more: "You don't need this",
    },
  },
];

export const formattedBooksArray = [
  {
    title: "Tom's Test Literature",
    authors: 'Chris, Mary',
    publisher: "Tom's Books",
    publishedDate: '2020-06-11',
  },
  {
    title: "Tom's SAT Prep",
    authors: 'Chris, Mary',
    publisher: "Tom's Books",
    publishedDate: '2020-06-11',
  },
];
