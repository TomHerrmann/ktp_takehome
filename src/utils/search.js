const search = (query, booksArr) => {
  const results = [];

  for (let i = 0; i < storeArr.length; i++) {
    if (
      booksArr[i].title.includes(query.toLowerCase()) ||
      booksArr[i].authors.includes(query.toLowerCase()) ||
      booksArr[i].publisher.includes(query.toLowerCase())
    )
      results.push(storeArr[i]);
  }

  return results;
};

export default search;
