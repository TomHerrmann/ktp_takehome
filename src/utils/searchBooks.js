const searchBooks = (query, booksArr) => {
  const results = [];

  for (let i = 0; i < booksArr.length; i++) {
    if (
      booksArr[i].title.toLowerCase().includes(query.toLowerCase()) ||
      booksArr[i].authors.toLowerCase().includes(query.toLowerCase()) ||
      booksArr[i].publisher.toLowerCase().includes(query.toLowerCase())
    )
      results.push(booksArr[i]);
  }

  return results;
};

export default searchBooks;
