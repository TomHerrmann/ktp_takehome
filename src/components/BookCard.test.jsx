import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import BookCard from './BookCard.jsx';
import { formattedBooksArray } from '../utils/enums';

describe('<CharacterCard />', () => {
  let defaultProps;
  beforeEach(() => {
    defaultProps = {
      book: formattedBooksArray[0],
    };
  });

  it('should render correctly', () => {
    const component = shallow(<BookCard {...defaultProps} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
