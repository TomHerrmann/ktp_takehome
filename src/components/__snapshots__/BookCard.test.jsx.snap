// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CharacterCard /> should render correctly 1`] = `
<article
  className="book-card"
>
  <h2>
    Tom's Test Literature
  </h2>
  <ul>
    <li
      key="0Chris, Mary"
    >
      <strong>
        Author(s): 
      </strong>
      Chris, Mary
    </li>
    <li
      key="1Tom's Books"
    >
      <strong>
        Publisher: 
      </strong>
      Tom's Books
    </li>
    <li
      key="22020-06-11"
    >
      <strong>
        Publish Date: 
      </strong>
      2020-06-11
    </li>
  </ul>
</article>
`;
